hail/python/hail/backend/service_backend.py

```diff
@@ -214,6 +215,12 @@ async def create(*,
         worker_memory = configuration_of('query', 'batch_worker_memory', worker_memory, None)
         name_prefix = configuration_of('query', 'name_prefix', name_prefix, '')
 
+        regions_str = configuration_of('batch', 'regions', regions, None)
+        if regions_str is not None:
+            regions = regions_str.split(',')
+        else:
+            regions = bc.supported_regions()
```

Comment:
The regions parameter below is `Optional[List[str]]`, but from this else case it looks like we don't exercise the `None` option. I don't have a strong opinion about whether to do one or the other, but I think we should either make `regions` not Optional, or keep this None instead of using `supported_regions`.

Reply:
I think we should try and keep the same interface as `hailtop.batch` as much as possible for consistency reasons. This would mean we'd need to expose the `ServiceBackend.supported_regions` and the equivalent of `ServiceBackend.ANY_REGION`. I think they should be `hl.supported_regions()` that is implemented as a static method on the service backend when applicable and `hl.ANY_REGION` is just an object. Once we figure this out, then we can figure out what makes sense for the types further downstream.

Reply:
> I think we should try and keep the same interface as hailtop.batch as much as possible for consistency reasons.

Totally agree with this, but I see the constructor of the `ServiceBackend` object and its implementation as private, so I don't see why we can't currently write the type as it's being used -- if it's never `None` let's use that knowledge to cut down on None checks that are currently dead code.

Reply:
I agree that the user-facing API should be as close to batch as possible, but the `ServiceBackend` for query is not a user-facing API. Users only use `hl.init` which select the backend to use.